extensions/html/server/src/modes/pathCompletion.ts

```diff
@@ -0,0 +1,65 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import { TextDocument, Position, CompletionList, CompletionItemKind } from 'vscode-languageserver-types';
+import { WorkspaceFolder } from 'vscode-languageserver-protocol/lib/protocol.workspaceFolders.proposed';
+import * as path from 'path';
+import * as fs from 'fs';
+import uri from 'vscode-uri';
+
+export function getPathCompletionParticipant(
+	document: TextDocument,
+	position: Position,
+	result: CompletionList,
+	workspaceFolders: WorkspaceFolder[] | undefined
+) {
+	return {
+		onHtmlAttributeValue: (tag, attributeName, attributeValue) => {
+			const pathTagAndAttribute: { [t: string]: string } = {
```

Comment:
better declare outside of function

Reply:
Compiled a list of them from https://github.com/Microsoft/vscode/issues/2037#issuecomment-367785300